src/prevector.h

```diff
@@ -276,8 +278,13 @@ class prevector {
         return *this;
     }
 
-    prevector& operator=(prevector<N, T, Size, Diff>&& other) {
-        swap(other);
+    prevector& operator=(prevector<N, T, Size, Diff>&& other) noexcept {
+        if (!is_direct()) {
+            free(_union.indirect_contents.indirect);
+        }
+        _union = std::move(other._union);
+        _size = other._size;
```

Comment:
Maybe a good idea to:

```C++
if (!other.is_direct()) {
        other._union.indirect_contents.indirect = nullptr;
}
```

in case anyone tries to `free` it?

Reply:
I think it's enough to have `other._size = 0;` because then `other.is_direct()` will return false and no free happens